[oneTBB] Add the Core Types preference API to the spec (#394)
oneapi-src/oneAPI-spec: source/elements/oneTBB/source/task_scheduler/task_arena/task_arena_cls.rst

                };

                struct constraints {
                    numa_node_id numa_node;
                    int max_concurrency;

                    constraints(numa_node_id numa_node_       = task_arena::automatic,

>                                 int          max_concurrency_ = task_arena::automatic);

Author: @alexey-katranov @aleksei-fedotov 

Folks, how do you think, do we want to remove this constructor from the spec?

For me, it seems like the removing makes sense, but it breaks the source backward compatibility.

Author: It was decided that the constructor should exist for all pre-C++20 standards. C++20 standard and later shouldn't have this constructor.

>     Represents limitations applied to threads within ``task_arena``.
> 

Author: ```suggestion
    Represents limitations applied to threads within ``task_arena``.

    Starting from C++20 this class should be an aggregate type to support the designated initialization.

```

.. cpp:member:: numa_node_id constraints::numa_id

    An integral logical index uniquely identifying a NUMA node.
    If set to non-automatic value, then this NUMA node will be considered as preferred for all the
    threads within the arena.

    .. note::

        NUMA node ID is considered valid if it was obtained through tbb::info::numa_nodes().

.. cpp:member:: int constraints::max_concurrency

    The maximum number of threads that can participate in work processing
    within the ``task_arena`` at the same time.

.. cpp:member:: core_type_id constraints::core_type

    An integral logical index uniquely identifying a core type.
    If set to non-automatic value, then this core type will be considered as preferred for all the
    threads within the arena.

    .. note::

        core type ID is considered valid if it was obtained through ``tbb::info::core_types()``.

.. cpp:member:: int constraints::max_threads_per_core

    The maximum number of threads that can be scheduled to one core simultaneously.

.. cpp:function:: constraints::constraints(numa_node_id numa_node_ = task_arena::automatic, int max_concurrency_ = task_arena::automatic)

>     Constructs the constriants object with the provided `numa_id` and `max_concurrency` settings.

Reviewer: A typo:
```suggestion
    Constructs the constraints object with the provided `numa_id` and `max_concurrency` settings.
```

    .. note::

>         Starting from C++20 this constructor does not exist.

Author: ```suggestion
        To support designated initialization this constructor is omitted starting from C++20. Aggregate initialization is supposed to be used instead.
```

>         To allow using the C++20 designated initialization the constructor is removed for the C++20 and later standard.

Reviewer: I don't think we need to explain the reasons here.
Can we just say:
```suggestion
        Starting from C++20 this constructor does not exist.
```
?

    ``max_concurrency`` - The maximum number of threads that can participate in work processing
    within the ``task_arena`` at the same time.

    ``core_type`` - An integral logical index uniquely identifying a core type.
    If set to non-automatic value, then this core type will be considered as preferred for all the
    threads within the arena.

    .. note::

        core type ID is considered valid if it was obtained through ``tbb::info::core_types()``.

>     ``core_type`` - The maximum number of threads that can be scheduled to one core simultaneously.

Reviewer: A typo:
```suggestion
    ``max_threads_per_core`` - The maximum number of threads that can be scheduled to one core simultaneously.
```
